### ISSUE TITLE
When migrating a credit results in multiple hits, provide a UI to pick which one of the hits is correct

### ISSUE THREAD

**Issue by LegalizeAdulthood** [GrandComicsDatabase/gcd-django#546]
Currently what happens is that the credit is unmigrated and you manually copy/paste the creators name into the box and pick from one of the database hits via the dropdown.

Since we already did the database search and found the multiple hits when we attempted to migrate the credit, provide some sort of UI that lists the creators that were found to let you fill out the field with a one-click step using front-end javascript and no additional server round trip.

**Comment by LegalizeAdulthood** [comment]
... the same sort of UI could be provided for features that had multiple hits, e.g. Captain Atom.

**Comment by jochengcd** [comment]
Would be a new database search (although maybe a cached query by the db-server). There is no practical path to further propagate intermediate results from the migration back to the UI. Only by storing these "somewhere" intermediately, would likely would be make the flow more state-dependent and brittle.

Nice idea, but I think too little gain for the needed work.

**Comment by LegalizeAdulthood** [comment]
Can you point me to where the migration algorithm is in the code?  I'll take a look and see if there's some potential "low hanging fruit".

**Comment by jochengcd** [comment]
The migration is on apps/oi/models.py for the issue and story revision model. 

To pass the info forward one would also need to change the routines (and forms ?) for bringing the editing page. Which would then become state dependent, which we try to avoid ?

**Comment by LegalizeAdulthood** [comment]
Can you elaborate on what you mean by "state dependent"?

**Comment by jochengcd** [comment]
The front end editing would depend on how one gets there. There would be a difference between reaching the editing page by clicking edit vs. after a migration click. The code for making the form and page would know to know this state.

**Comment by LegalizeAdulthood** [comment]
I'm actually thinking that the entire thing can be provided in the front-end... there is already some sort of ajax discussion going on between the form and the back end for the dynamic search when you are typing in the name of a creator.  It shouldn't matter whether you get there by 'edit' or by 'migrate'.

**Comment by jochengcd** [comment]
Sure. But then there is the question if this should be there for every unmigrated credit.We don't force indexers to migrate, so it should be only there on request. In other words, for which text credits should this show ? In that view, it would make sense to show this only after a migrate ? Or do provide a button for an 'in page' migration for each text credit ?

**Comment by LegalizeAdulthood** [comment]
My thinking was that it would appear below the credits text box as openable triangle (that is initially closed) with some sort of text like "pick one from a list".  Opening the triangle would result in choices appearing below the text credit box and picking one would result in the equivalent of "add another" being clicked and the appropriate fields being filled out.

**Comment by LegalizeAdulthood** [comment]
My thinking was that there would be a single (initially collapsed) triangle altering you to the availability of autocomplete "pick and migrate" options and that expanding it would show you all the relevant possibilities for that credit.

You're right that if there are a bunch of creators listed for a single credit, then it could balloon into a large list of options to choose from -- one pickable result for each autocomplete response provided for each creator.  So I think it's best to have this hidden by default.

There's also the situation that we may not like the script running all these autocomplete queries every time you edit an issue and the search could be delayed until you expand the triangle manually.

So far, everything I've described is achievable only with front-end JavaScript code.

**Comment by J-Garcke-SCAI** [comment]
Yes, checking if if there are autocompletes shouldn't be done for each story editing page, would be quite a bit of unneeded queries.

But then, if it is up to the user clicking on the triangle manually, than there is overall not so much difference to clicking the migrate button for the whole issue ? (as in, for the text credits that are unmigrated since nothing was done so far, on them, not for the unmigrated due to multiple hits)

Maybe not that straightforward ? Thinking this through gives a bit of side constraints / effects ?

**Comment by jochengcd** [comment]
Yes, checking if if there are autocompletes shouldn't be done for each story editing page, would be quite a bit of unneeded queries.

But then, if it is up to the user clicking on the triangle manually, than there is overall not so much difference to clicking the migrate button for the whole issue ? (as in, for the text credits that are unmigrated since nothing was done so far, on them, not for the unmigrated due to multiple hits)

Maybe not that straightforward ? Thinking this through gives a bit of side constraints / effects ?

**Comment by LegalizeAdulthood** [comment]
The idea is to help automate the process that isn't handled by the existing migrate button.  Maybe we can do something so that this is only rendered into the form on the result of a 'migrate credits' submission and not on an ordinary edit submission.
